haystack/preview/components/audio/whisper_remote.py

```diff
@@ -30,108 +27,116 @@ class RemoteWhisperTranscriber:
+            **self.whisper_params,
+        )
 
-        :param audio_files: a list of paths or binary streams to transcribe
-        :returns: a list of transcriptions.
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "RemoteWhisperTranscriber":
         """
-        transcriptions = self._raw_transcribe(audio_files=audio_files, **kwargs)
-        documents = []
-        for audio, transcript in zip(audio_files, transcriptions):
-            content = transcript.pop("text")
-            if not isinstance(audio, (str, Path)):
-                audio = "<<binary stream>>"
-            doc = Document(text=content, metadata={"audio_file": audio, **transcript})
-            documents.append(doc)
-        return documents
+        Deserialize this component from a dictionary.
+        """
+        return default_from_dict(cls, data)
 
-    def _raw_transcribe(self, audio_files: Sequence[Union[str, Path, BinaryIO]], **kwargs) -> List[Dict[str, Any]]:
+    @component.output_types(documents=List[Document])
+    def run(self, streams: List[ByteStream]):
         """
-        Transcribe the given audio files. Returns a list of strings.
+        Transcribe the audio files into a list of Documents, one for each input file.
 
         For the supported audio formats, languages, and other parameters, see the
         [Whisper API documentation](https://platform.openai.com/docs/guides/speech-to-text) and the official Whisper
         [github repo](https://github.com/openai/whisper).
 
-        :param audio_files: a list of paths or binary streams to transcribe.
-        :param kwargs: any other parameters that Whisper API can understand.
-        :returns: a list of transcriptions as they are produced by the Whisper API (JSON).
+        :param audio_files: a list of ByteStream objects to transcribe.
+        :returns: a list of Documents, one for each file. The content of the document is the transcription text.
         """
-        translate = kwargs.pop("translate", False)
-        url = f"{self.api_base}/audio/{'translations' if translate else 'transcriptions'}"
-        data = {"model": self.model_name, **kwargs}
-        headers = {"Authorization": f"Bearer {self.api_key}"}
-
-        transcriptions = []
-        for audio_file in audio_files:
-            if isinstance(audio_file, (str, Path)):
-                audio_file = open(audio_file, "rb")
-
-            request_files = ("file", (audio_file.name, audio_file, "application/octet-stream"))
-            response = request_with_retry(
-                method="post", url=url, data=data, headers=headers, files=[request_files], timeout=OPENAI_TIMEOUT
-            )
-            transcription = json.loads(response.content)
+        documents = []
 
-            transcriptions.append(transcription)
-        return transcriptions
+        for stream in streams:
+            try:
+                file = io.BytesIO(stream.data)
+                file.name = stream.metadata["file_path"]
```

Comment:
@awinml yes, let's do a check here if stream.metadata["file_path"] is present. If it is, use it. If not, just use a random name, e.g. `audio_input` , perhaps an extension is not even needed. Please check.

Reply:
I tried the API without using an extension, it does not allow that. I think, we can use `audio_input.wav` as the name if `stream.metadata["file_path"]` is not present.

I updated the [example notebook](https://colab.research.google.com/drive/1FlxKkw3Q83k7Z-BfXPYat5UDt_iA5rIO?usp=sharing) with a test.

Reply:
I wouldn't. That file name is not important at all, as far as I can tell, and we needlessly scare users. wdyt @ZanSara @awinml

Reply:
Makes sense, I will push the changes without the warning then.